docs/releasing.md

```diff
@@ -0,0 +1,138 @@
+# Releasing ironic-image
+
+This document explains how to prepare and publish a release for ironic-image, and
+how often a new release branch is created.
+
+**NOTE**: Always follow
+[release documentation from the main branch](https://github.com/metal3-io/ironic-image/blob/main/docs/releasing.md).
+Release documentation in release branches may be outdated.
+
+## Pre-release actions
+
+Things you should check before making a release:
+
+- Check the
+  [Metal3 release process](https://github.com/metal3-io/metal3-docs/blob/main/processes/releasing.md)
+  for high-level process and possible follow-up actions
+- Verify the latest bugfix or stable branches (which is the most recent) in ironic
+  and ironic-inspector upstream repositories;
+- Verify compatibility with latest sushy and ironic-lib upstream releases or master
+  branches based on ironic and ironic-inspector bugfix or stable requirements
+  and constraints
+- Verify openstack upper-constraints compatibility with ironic and ironic-inspector
+  bugfix or stable branches
+- Verify any other direct or indirect dependency is uplifted to close any public
+  vulnerabilities
+
+  ## Permissions
+
+Creating a release requires repository `write` permissions for:
+
+- Tag pushing
+- Branch creation
+- GitHub Release publishing
+
+These permissions are implicit for the org admins and repository admins.
+A release team member gets permissions via `metal3-release-team`
+membership. This GitHub team has the required permissions in each repository
+to make a release. Adding person to the team provides the necessary
+rights  in all relevant repositories in the organization. Individual persons
+should not be given permissions directly.
+
+## Process
+
+ironic-image uses [semantic versioning](https://semver.org). For version `vX.x.y`:
+
+### Repository setup
+
+Clone the repository:
+`git clone git@github.com:metal3-io/ironic-image`
+
+or if using existing repository, verify your intended remote is set to
+`metal3-io`: `git remote -v`. For this document, we assume it is `origin`.
+
+- If creating a new release branch, identify the commit you wish to create the
+  branch from, and create a branch `release-X.x` where `X.x` corresponds to the
+  ironic upstream release version, for example for `ironic 22.1` it's `release-22.1`:
+  `git checkout <sha> -b release-X.x` and push it to remote:
+  `git push origin release-X.x` to create it
+- If creating a new patch release, use existing branch `release-X.x`:
+  `git checkout origin/release-X.x`
+- In the newly created release branch:
+   - Add a copy of upper-constraints.txt from [Openstack Requirements](https://opendev.org/openstack/requirements)
+     to replace the placeholder
+   - Pin ironic and ironic-inspector to match the corresponding bugfix or
```

Comment:
Ironic only tests upgrades to/from major 6 month releases. In addition, the stable 6 month releases are supported for a long life cycle -- guaranteed for 18 months -- where bugfix branches have a much shorter lifecycle.

Additionally, there is minimal value in metal3 using Ironic bugfix branches as it's relatively rare that Ironic has a new feature that metal3 immediately consumes. 

What is the primary driver behind tracking bugfix releases in Ironic?

Reply:
@jayofdoom thanks for the comment!
we're not going to always tie the release/branch of the ironic-image to bugfix branches, we'll use either bugffix or stable based on which is the closest one as a general tule of thumb
the main reason why we can use the bugfix branches, considering their short life, is that the release/branches of the ironic-image will also be supported for no longer than 6 months, except very special cases
we aim to branch 3-4 times a year, meaning every 3-4 months (shorter than lifetime of a bugfix branch), and release maybe even more often

Reply:
> it's relatively rare that Ironic has a new feature that metal3 immediately consumes.

This is not quite true, by the way. We have several features developed specifically for Metal3 consumption. This includes JSON RPC or all-in-one Ironic executable in the past, firmware upgrades and non-bootable ISO presently.

Reply:
Ack. I think part of why this isn't ideal is that bugfix releases are not handled entirely via automation on the Ironic side, so it makes me nervous to have another OSS downstream consumer of it -- but frankly, you both know I've felt that way about them since the beginning.

Reply:
To close the loop here: Riccardo is looking into improving the automation.